refactor(migrations): add internal cleanup logic

crisbeto opened the pull request:
Expands the `inject` migration to add some cleanups that are only relevant internally. Externally this isn't exposed to users.

devversion commented on packages/core/schematics/ng-generate/inject-migration/internal.ts:
```diff
@@ -0,0 +1,202 @@
+      );
+
+      if (isReferencingLocalSymbol) {
+        hasLocalRefs = true;
```
Should we skip visiting children if we determined `hasLocalRefs = true`?

crisbeto committed:
refactor(migrations): add internal cleanup logic

Expands the `inject` migration to add some cleanups that are only relevant internally. Externally this isn't exposed to users.

AndrewKushnir commented:
This PR was merged into the repository by commit 05c3750b083fd0eddf646700059db6c68db048ab.

The changes were merged into the following branches: main, 18.2.x
